doc/api/async_hooks.md

```diff
@@ -712,6 +712,88 @@ never be called.
 * Returns: {number} The same `triggerAsyncId` that is passed to the
 `AsyncResource` constructor.
 
+## Class: AsyncContext
+<!-- YAML
+added: REPLACEME
+-->
+
+This class is used to create asynchronous state between callbacks and promise
+chains. It allows you to keep variables throughout the lifetime of a web request
+or any other asynchronous duration. It is similar to what ThreadLocal storage is
+ in other languages.
+
+### Example
+
+Let's build a logger that will always know the current HTTP request and use
```

Comment:
The API documentation should avoid language like "Let's build a logger." Instead, maybe go with "The following example builds a logger..."